packages/wdio-appium-service/src/launcher.ts

```diff
@@ -173,8 +173,8 @@ export default class AppiumLauncher implements Services.ServiceInstance {
             } else if (error) {
                 errorMessage += `\n${error.toString()}`
             }
-            log.error(errorMessage)
             callback(new Error(errorMessage), null)
+            exitCode !== 0 ? log.error(errorMessage) : ''
```

Comment:
Let's log this before the callback is called

Reply:
and do:
```ts
if (exitCode !== 0) {
    log.error(errorMessage)
}
```

Reply:
rewritten